e2e/test/scenarios/question/column-compare.cy.spec.ts

```diff
@@ -10,134 +10,798 @@ import {
+  popover()
+    .findByText(/compare/)
+    .should("not.exist");
+}
+
+type CheckTextOpts = {
+  itemName: string;
+  step1Title?: string;
+  step2Title: string;
+  offsetHelp: string;
+};
+
+function verifySummarizeText(options: CheckTextOpts) {
+  cy.button("Summarize").click();
+  rightSidebar().button("Add aggregation").click();
+
+  popover().within(() => {
+    cy.findByText(options.itemName).should("exist").click();
+
+    if (options.step1Title) {
+      cy.findByText(options.step1Title).should("exist");
+      cy.findByText("Sum of Price").should("exist");
+      cy.findByText("Count").click();
+    }
+
+    cy.findByText(options.step2Title).should("exist");
+    cy.findByText(options.offsetHelp).should("exist");
+  });
+}
+
+function verifyColumnDrillText(options: Omit<CheckTextOpts, "step1Title">) {
+  tableHeaderClick("Count");
+
+  popover().within(() => {
+    cy.findByText(options.itemName).should("exist").click();
+    cy.findByText(options.step2Title).should("exist");
+    cy.findByText(options.offsetHelp).should("exist");
+  });
+}
+
+function verifyPlusButtonText(options: CheckTextOpts) {
+  cy.button("Add column").click();
+
+  popover().within(() => {
+    cy.findByText(options.itemName).should("exist").click();
+
+    if (options.step1Title) {
+      cy.findByText(options.step1Title).should("exist");
+      cy.findByText("Sum of Price").should("exist");
+      cy.findByText("Count").click();
+    }
+
+    cy.findByText(options.step2Title).should("exist");
+    cy.findByText(options.offsetHelp).should("exist");
+  });
+}
+
+function verifyNotebookText(options: CheckTextOpts) {
+  openNotebook();
+  getNotebookStep("summarize")
+    .findByTestId("aggregate-step")
+    .icon("add")
+    .click();
+
+  popover().within(() => {
+    cy.findByText(options.itemName).should("exist").click();
+
+    if (options.step1Title) {
+      cy.findByText(options.step1Title).should("exist");
+      cy.findByText("Sum of Price").should("exist");
+      cy.findByText("Count").should("exist").click();
+    }
+
+    cy.findByText(options.step2Title).should("exist");
+    cy.findByText(options.offsetHelp).should("exist");
+  });
+}
+
+type AggregationResult = {
+  name: string;
+  expression: string;
+};
+
+function verifyAggregations(results: AggregationResult[]) {
+  for (const result of results) {
+    cy.findByTestId("aggregate-step")
+      .findByText(result.name)
+      .should("exist")
+      .click();
+
+    cy.get(".ace_content").should("have.text", result.expression);
+
+    cy.realPress("Escape");
+  }
+}
+
+function verifyColumns(names: string[]) {
+  visualize();
+
+  for (const name of names) {
+    cy.findAllByTestId("header-cell").contains(name).should("exist");
+  }
+}
+
+function verifyBreakoutRequiredError() {
+  visualize();
+
+  cy.get("main")
+    .findByText("There was a problem with your question")
+    .should("exist");
+  cy.get("main").findByText("Show error details").should("exist").click();
```

Comment:
nit: both `findByText` and `click` make internal assertion and will fail if the element doesn't exist
```suggestion
  cy.get("main").findByText("Show error details").click();
```

Reply:
Usually I want to be explicit that it's an intentional assertion and not a part of "test flow", but yeah, it does not make much sense here: 836e13e7af111c70ef1b1f6e224634815c342cfd